### DISCUSSION TITLE
Is there any way to abort previous RTKQ request?

### DISCUSSION THREAD

**Discussion by Leeasdw** [reduxjs/redux-toolkit#4076] [Q&A] [answered]
I have to show request time on the page.  For now I'm using something like : 
```JavaScript
    const startTime = new Date().getTime();
    //await lazy RTKQ request
    const endTime = new Date().getTime();
    const duration = (endTime - startTime) / 1000
```
and then put this in useState. (since I haven't found a better way to get request time).

Everything is okay, but if the previous request finishes after the last one, then it overwrites the time also. 
Data isn't being overwritten, because as far I understand it's managed by RTKQ.

So, is there any way to cancel previous request or maybe different approach on getting request time, so I can avoid this problem? 

Would be grateful for any help!

**Comment by phryneas** [comment]
`useQuery` returns `startedTimeStamp` and `fulfilledTimeStamp` - do those work for you?

**Reply by Leeasdw** [reply]
Yeah, they do. It's actually a better way to count, thanks. But still, it doesn't solve the problem with overwriting :(

**Reply by Leeasdw** [reply, marked as answer]
nvm, I was doing the wrong thing. I was taking timestamps from lazy trigger method and then write it to the state. Instead I needed to take it from useQuery directly, as you said. Then rtkq manages them correctly by itself
